esx_service/vmdk_ops.py

```diff
@@ -170,55 +169,59 @@ def createVMDK(vmdk_path, vm_name, vol_name, opts={}, vm_uuid=None, tenant_uuid=
         return cloneVMDK(vm_name, vmdk_path, opts,
                          vm_uuid, datastore)
 
-    cmd = make_create_cmd(opts, vmdk_path)
-    rc, out = RunCommand(cmd)
-    if rc != 0:
-        return err("Failed to create %s. %s" % (vmdk_path, out))
+    if not kv.DISK_ALLOCATION_FORMAT in opts:
+        disk_format = kv.DEFAULT_ALLOCATION_FORMAT
+    else:
+        disk_format = kv.VALID_ALLOCATION_FORMATS[opts[kv.DISK_ALLOCATION_FORMAT]]
+
+    # VirtualDiskSpec
+    vdisk_spec = vim.VirtualDiskManager.FileBackedVirtualDiskSpec()
+    vdisk_spec.adapterType = VMDK_ADAPTER_TYPE
+    vdisk_spec.diskType = disk_format
+
+    if  kv.SIZE in opts:
+        vdisk_spec.capacityKb = convert.convert_to_KB(opts[kv.SIZE])
+    else:
+        vdisk_spec.capacityKb = convert.convert_to_KB(kv.DEFAULT_DISK_SIZE)
+
+    # Form datastore path from vmdk_path
+    volume_datastore_path = vmdk_utils.get_datastore_path(vmdk_path)
+
+    si = get_si()
+    task = si.content.virtualDiskManager.CreateVirtualDisk(
```

Comment:
Can we please file a tracking issue to investigate s/vim.VirtualDiskManager/ vim.vslm.host.VStorageObjectManager/ APIs.